### ISSUE TITLE
SA1118 triggers on C# 11 raw string literals

### ISSUE THREAD

**Issue by alexrp** [DotNetAnalyzers/StyleCopAnalyzers#3540]
![image](https://user-images.githubusercontent.com/44076/181410113-785f5d73-12e4-4d41-87ec-d6ab138bad70.png)

**Comment by sharwell** [comment]
As best I can tell, this warning is also reported for multi-line verbatim string literals.

**Comment by alexrp** [comment]
It seems reasonable that such strings should be added to the list of exceptions here: https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1118.md#rule-description

**Comment by sharwell** [comment]
I believe the current behavior is in line with the intent of SA1118 (specifically, it is difficult to quickly see where one argument ends and the next begins). Based on this, I would say it's unlikely for a change to be made here.

Note that a DiagnosticSuppressor is always an option if you want to keep a rule but not have it interfere with one specific use case.

**Comment by alexrp** [comment]
Ok, will close this then.

**Comment by JoC0de** [comment]
With this rule also applying to interpolated strings there is no way to write code like:
```csharp
var jsonRequest = string.Create(
   CultureInfo.InvariantCulture,
   $$"""
   {
      "id": "{{person.Id}}",
      "name": "{{person.Name}}"
   }
   """);
```
The signature of `string.Create` is: 
```
public static string Create (IFormatProvider? provider, ref System.Runtime.CompilerServices.DefaultInterpolatedStringHandler handler);
```
As `System.Runtime.CompilerServices.DefaultInterpolatedStringHandler` is a `ref struct` there is no "compiling" way of creating a temporary variable and parsing it to the `string.Create` inside async code. So the only option is to ignore the warning for this cases.

**Comment by sharwell** [comment]
@JoC0de would make sense to file a new issue specifically for the case of interpolated strings getting passed to an interpolated string handler.
